spellbook/models/sudoswap/ethereum/sudoswap_ethereum_events.sql

```diff
@@ -0,0 +1,240 @@
+{{ config(
+        alias ='events',
+        materialized ='incremental',
+        file_format ='delta',
+        incremental_strategy='merge',
+        unique_key='unique_trade_id'
+        )
+}}
+
+--base table CTEs
+WITH 
+    swaps as (
+        SELECT 
+            * 
+        FROM (
+            SELECT 
+                contract_address
+                , call_tx_hash
+                , call_trace_address
+                , call_block_time
+                , call_block_number
+                , call_success
+                , tokenRecipient as call_from
+                , 'Sell' as trade_category 
+            FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_call_swapNFTsForToken') }}
+            UNION ALL
+            SELECT 
+                contract_address
+                , call_tx_hash
+                , call_trace_address
+                , call_block_time
+                , call_block_number
+                , call_success
+                , nftRecipient as call_from
+                , 'Buy' as trade_category 
+            FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_call_swapTokenForAnyNFTs') }}
+            UNION ALL
+            SELECT
+                contract_address
+                , call_tx_hash
+                , call_trace_address
+                , call_block_time
+                , call_block_number
+                , call_success
+                , nftRecipient as call_from
+                , 'Buy' as trade_category 
+            FROM {{ source('sudo_amm_ethereum','LSSVMPair_general_call_swapTokenForSpecificNFTs') }}
+        ) s
+        {% if is_incremental() %}
```

Comment:
we've found it beneficial performance wise to move these incremental filters to within the CTEs -- for example, you'd apply this if block and filter to all three queries in the swaps CTE